transport_analysis/velocityautocorr.py

```diff
@@ -262,3 +261,68 @@ def plot_vacf(self, start=0, stop=0, step=1):
             self.times[start:stop:step],
             self.results.timeseries[start:stop:step],
         )
+
+    def sd(self, start=0, stop=0, step=1):
+        """
+        Returns a self-diffusivity value using ``scipy.integrate.trapezoid``.
+        Analysis must be run prior to computing self-diffusivity.
+
+        Parameters
+        ----------
+        start : Optional[int]
+            The first frame of ``self.results.timeseries``
+            used for the calculation.
+        stop : Optional[int]
+            The frame of ``self.results.timeseries`` to stop at
+            for the calculation, non-inclusive.
+        step : Optional[int]
+            Number of frames to skip between each frame used
+            for the calculation.
+
+        Returns
+        -------
+        `numpy.float64`
+            The calculated self-diffusivity value for the analysis.
+        """
+        stop = self.n_frames if stop == 0 else stop
```

Comment:
Should raise an exception if .run hasn't already been called with some kind of cache variable.

Reply:
I'm not too sure where to start here. Do you think you'd be able to explain a bit more, or would you be good with setting up a meeting to go over this?

Reply:
Yep happy to meet. 

Something like the following should work. 


```python

def __init__(self,)
    self._run_called = False
   ...

   def _conclude(self):
   self._run_called = True
   ...

   def sd_gk(self):
    if not self._run_called:
        raise RunTimeError("blah blah")
```

Reply:
Or `self._has_run`?

Something like, `"VelocityAutocorrelation.run() must be called before 'sd_gk'"`